pom.xml

```diff
@@ -308,6 +308,34 @@ limitations under the License.
           </excludes>
         </configuration>
       </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-checkstyle-plugin</artifactId>
+        <version>3.1.2</version>
+        <configuration>
+            <configLocation>.mvn/checkstyle.xml</configLocation>
+            <encoding>UTF-8</encoding>
+            <consoleOutput>true</consoleOutput>
+            <failsOnError>true</failsOnError>
+            <includeTestSourceDirectory>true</includeTestSourceDirectory>
+        </configuration>
+        <dependencies>
+          <dependency>
+            <groupId>com.puppycrawl.tools</groupId>
+            <artifactId>checkstyle</artifactId>
+            <version>9.3</version>
+          </dependency>
+        </dependencies>
+        <executions>
+          <execution>
+            <id>validate</id>
+            <phase>validate</phase>
```

Comment:
(Opinion) I don't like including it by default in a phase, it will be redundant for every build to use it.  It really only needs to be run once in a project, and putting it in the validate phase executes it for every build.  Just include `checkstyle:check` in one of the GHA configurations.

Reply:
I personally prefer to have a default build containing all the checks, but what I use to do is a profile activated with `-Dfast`, this profile would deactivate some plugins like javadoc, jacoco, checkstyle, etc so that a `./mvnw clean install -Dfast` is... fast.

I think a default build should be done in a way to include all the steps that should be included by default but any hack to improve dev time should be activated by the devs. Not the way around.

Reply:
(and such profile and dev improvement can be done in another PR)